worktree_commit.go

```diff
@@ -189,7 +190,21 @@ func (h *buildTreeHelper) BuildTree(idx *index.Index, opts *CommitOptions) (plum
 		}
 	}
 
-	return h.copyTreeToStorageRecursive(rootNode, h.trees[rootNode])
+	treeHash, err := h.copyTreeToStorageRecursive(rootNode, h.trees[rootNode])
+	if err != nil {
+		return plumbing.ZeroHash, err
+	}
+
+	parentHash := plumbing.ZeroHash
+	if len(opts.Parents) > 0 {
+		parentHash = opts.Parents[0]
+	}
+
+	if treeHash == parentHash && !allowEmpty {
```

Comment:
We should compare tree hashes against each other (current vs parent). Here we are comparing the tree hash against the parent commit hash.

Please change that, but add this check on the calling method instead.